src/GHC/Util.hs

```diff
@@ -288,3 +297,25 @@ getloc = getLoc
 
 noext :: NoExt
 noext = noExt
+
+--
+
+newtype SrcSpanD = SrcSpanD SrcSpan -- SrcSpan, deriving `Default`.
+instance Default SrcSpanD where def = SrcSpanD noSrcSpan
+instance Outputable SrcSpanD where ppr (SrcSpanD s) = ppr s
+instance Eq SrcSpanD where (==) (SrcSpanD a) (SrcSpanD b) = a == b
```

Comment:
Can this use `GeneralizedNewtypeDeriving`? Should work for all but Default.

Reply:
I'd have to research that. Perhaps you can give me a pointer?

Reply:
`newtype SrcSpanD = SrcSpanD SrcSpan deriving (Outputable, Eq, Ord)` and add `GeneralizedNewtypeDeriving` at the top of the file when the compiler yells at you.

Reply:
Ok!

Reply:
✔️